eml/screens/lectures/VideoLectureScreen.jsx

```diff
@@ -0,0 +1,169 @@
+import React, { useState, useEffect, useRef } from 'react';
+
+// For animating play button
+
+import { Alert, View, Pressable } from 'react-native';
+
+import Text from '../../components/general/Text';
+
+import VideoActions from '../../components/lectures/VideoActions';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+
+
+import CustomExpoVideoPlayer from '../../components/lectures/VideoPlayer';
+import ReactSliderProgress from './ReactSliderProgress';
+import { getVideoDownloadUrl } from '../../api/api';
+
+import { useNavigation } from '@react-navigation/native';
+import ProgressTopBar from './ProgressTopBar';
+import axios from 'axios';
+
+
+
+export default function VideoLectureScreen({ lecture, course }) {
+
+
+    const videoRef = useRef(null);
+    const [isPlaying, setIsPlaying] = useState(true); // Keep track of playback status
+    const [positionMillis, setPositionMillis] = useState(0);
+    const [durationMillis, setDurationMillis] = useState(0);
+    const [isMuted, setIsMuted] = useState(false); // Keep track of mute status
+
+
+    const onStatusUpdate = (status) => {
+        setPositionMillis(status.positionMillis || 0);
+        setDurationMillis(status.durationMillis || 0);
+
+    };
+
+
+    const [videoUrl, setVideoUrl] = useState(null);
+
+    useEffect(() => {
+        const _videoUrl = getVideoDownloadUrl(lecture._id, "180p")
+
+        //send head request to check if video is valid
+        axios.head(_videoUrl).then((response) => {}).catch((error) => {
+            Alert.alert("Error", "The video is corrupted. Please try again later", "OK");
```

Comment:
You should consider pulling this sort of API out of screens section and into an API file, like the ones found in the API folder. 

I would also like to see the errors used (and in conjunction with that, the errorSwitch.js found in helpers). 
If you are unsure, you can find the ERRORS.md in resources. The idea is that all errors that is thrown in the backend and caught in the frontend should be following the ERROR guidelines.